cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/StatefulSetDiff.java

```diff
@@ -27,6 +27,7 @@
     private static final Pattern IGNORABLE_PATHS = Pattern.compile(
         "^(/spec/revisionHistoryLimit"
         + "|/spec/template/metadata/annotations/" + SHORTENED_STRIMZI_DOMAIN + "~1generation"
+        + "|/spec/template/metadata/annotations/" + SHORTENED_STRIMZI_DOMAIN + "~1old-logging-hash"
```

Comment:
Why do we need to carry this on the STS?

Reply:
I did not find any better way how to *not* roll the Kafka pods when the configuration was changed dynamically and keep the rolling update triggered when the logging has changed.

Reply:
Ok, maybe to put this differently: Why do we need to have this annotation if not to roll the pod when it changes? Normally we add these annotations exactly for the reason to roll the pod.

Reply:
I extract it here https://github.com/strimzi/strimzi-kafka-operator/pull/2389/files#diff-f19c7c3bdf294affc86939228666be84R1666
If the logging changed, we are not interested in dynamic configuration and we want to roll.

Reply:
But why? You do not change logging dynamically, or? Or did you included it in this PR?

Reply:
No, I am not changing it dynamically. Let's say, the Kafka CR has changed. Basically we can divide Kafka config into three parts. Metrics, logging and Kafka itself. Nowadays we are looking just at the logging at Kafka. With this PR we are changing Kafka part. We say it is possible we won't roll because the change was done dynamically. But if we change dynamically, the STS changes and the current algorithm says we need to roll. So we need to check whether it was dynamically changed. If it was we need to bypass the condition and do not roll. But if we change Kafka and logging part, it is not (currently) applicable, so we need to roll. So we changed Kafka config, the change is doable dynamically -> don't roll, but we have also logging -> roll.
This explanation is maybe more confusing, let me use pseudocode to try it better.

```
boolean shouldRoll;

// logging part
if (loggingChanged) {
 shouldRoll = true;
 // we do not care about Kafka config, we need to roll
 return;
}

// Kafka part
boolean kafkaChangedDynamically = changeKafkaConfig(old, new);
if (kafkaChangedDynamically) {
    shouldRoll = false;
}